include/neml2/models/Model.h

```diff
@@ -270,6 +270,19 @@ class Model : public NEML2Object, public LabeledAxisInterface, public NonlinearS
   */
   void register_model(std::shared_ptr<Model> model, bool merge_input = true);
 
+  /**
+   * Both register a model and return a reference
+   */
+  template <typename T, typename = typename std::enable_if_t<std::is_base_of_v<Model, T>>>
+  T & include_model(const std::string & name)
+  {
+    std::shared_ptr<Model> model = Factory::get_object_ptr<Model>("Models", name);
+
+    register_model(model, true);
```

Comment:
```suggestion
    register_model(model, merge_input);
```

as we'll retire `register_model` in the future.